frame/asset-rate/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+	#[pallet::storage]
+	pub type ConversionRateToNative<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AssetId, FixedU128, OptionQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		// Some `asset_id` conversion rate was created.
+		AssetRateCreated { asset_id: T::AssetId, rate: FixedU128 },
+		// Some `asset_id` conversion rate was removed.
+		AssetRateRemoved { asset_id: T::AssetId },
+		// Some existing `asset_id` conversion rate was updated from `old` to `new`.
+		AssetRateUpdated { asset_id: T::AssetId, old: FixedU128, new: FixedU128 },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The given asset ID is unknown.
+		UnknownAssetId,
+		/// The given asset ID already has an assigned conversion rate and cannot be re-created.
+		AlreadyExists,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Initialize a conversion rate to native balance for the given asset.
+		///
+		/// ## Complexity
+		/// - O(1)
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::create())]
+		pub fn create(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			rate: FixedU128,
+		) -> DispatchResult {
+			T::CreateOrigin::ensure_origin(origin)?;
+
+			ensure!(
+				!ConversionRateToNative::<T>::contains_key(asset_id),
+				Error::<T>::AlreadyExists
+			);
+			ConversionRateToNative::<T>::set(asset_id, Some(rate));
+
+			Self::deposit_event(Event::AssetRateCreated { asset_id, rate });
+			Ok(())
+		}
+
+		/// Update the conversion rate to native balance for the given asset.
+		///
+		/// ## Complexity
+		/// - O(1)
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::update())]
+		pub fn update(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			rate: FixedU128,
+		) -> DispatchResult {
+			T::UpdateOrigin::ensure_origin(origin)?;
+
+			let mut old = FixedU128::zero();
+			ConversionRateToNative::<T>::mutate(asset_id, |maybe_rate| {
+				if let Some(r) = maybe_rate {
+					old = *r;
+					*r = rate;
+
+					Ok(())
+				} else {
+					Err(Error::<T>::UnknownAssetId)
+				}
+			})?;
+
+			Self::deposit_event(Event::AssetRateUpdated { asset_id, old, new: rate });
+			Ok(())
+		}
+
+		/// Remove an existing conversion rate to native balance for the given asset.
+		///
+		/// ## Complexity
+		/// - O(1)
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::remove())]
+		pub fn remove(origin: OriginFor<T>, asset_id: T::AssetId) -> DispatchResult {
+			T::RemoveOrigin::ensure_origin(origin)?;
+
+			ensure!(
+				ConversionRateToNative::<T>::contains_key(asset_id),
+				Error::<T>::UnknownAssetId
+			);
+			ConversionRateToNative::<T>::remove(asset_id);
+
+			Self::deposit_event(Event::AssetRateRemoved { asset_id });
+			Ok(())
+		}
+	}
+}
+
+// Exposes conversion of an arbitrary balance of an asset to native balance.
+// TODO: Switch to `AssetBalanceToBalance` after merging Substrate #13610
```

Comment:
Can do now?

Reply:
Done in [58cc428](https://github.com/paritytech/substrate/pull/13608/commits/58cc428c9a06a3edb803195f945f7b669e0b9583)